### PR TITLE
Add --socket-timeout-ms option.

### PR DESCRIPTION
**Pull request by llvtt**
Addresses #185

### CODE REVIEW

**Comment by acmorrow** [comment]
FYI this is making the legacy C++ driver test integration tests red on Windows, and so is blocking our planned legacy-1.0.1-rc0

File: mongo_orchestration/common.py [hunk 1]
```diff
@@ -48,6 +49,7 @@ class BaseModel(object):
         {'role': 'dbAdminAnyDatabase', 'db': 'admin'},
         {'role': 'readWriteAnyDatabase', 'db': 'admin'}
     ]
+    socket_timeout = DEFAULT_SOCKET_TIMEOUT
```
Comment (behackett) [inline mongo_orchestration/common.py]: I don't really like this. Anywhere you call BaseModel.socket_timeout = <value> is going to change the value for all instances of subclasses. It's really inflexible.

Comment (llvtt) [inline mongo_orchestration/common.py]: The intention is that `socket_timeout` is consistent across Servers, ReplicaSets, and ShardedClusters. I'm not sure that there's a need to be able to set this property separately on each, and it's configured globally from the command-line anyway.

**Comment by behackett** [comment]
Ship it.

**Pull request merged**